cds-snc/covid-alert-app
# More granular error messages for OTC submission and TEK upload

# Summary

Closes #948. ~This does not cover logic for offline errors, just errors that depend on when response we get from the server.~ 

This PR now covers:
- One time code (OTC) submission errors:
  - offline
  - bad code
  - temporary ban
  - generic server error
- Temporary exposure key (TEK) upload errors:
  - offline
  - generic server error
  - didn't give permission

# Test instructions

1. checkout branch and create a build that talks to the staging server
2. generate a diagnosis key
3. enter a bad diagnosis key into the app - observe that it is now reading the `Error Code 2 = INVALID_ONE_TIME_CODE` error and displaying the "invalid code" error message
4. switch on airplane mode, try to submit, observe the offline error, switch off airplane mode
5. enter a good diagnosis key and observe that the app goes into the diagnoses state, as before
6. agree to upload your TEKs, then deny the system dialogue, observe the permission denied error message
7. switch on airplane mode, try to upload TEKs, observe the offline error, switch off airplane mode
8. agree to upload your TEKs, then accept the system dialogue, observe the app goes to the thank you page

To test the generic server error you can modify the app source code directly to throw a different server error code.

# Reviewer checklist

This is a suggested checklist of questions reviewers might ask during their review:
